### PR TITLE
Artifactory Release Lifecycle Management - Support creation by artifacts and sync for distribution

### PR DESCRIPTION
**Pull request by ns-kbhat**
- [ ] All [tests](https://github.com/jfrog/jfrog-client-go#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [ ] All [static analysis checks](https://github.com/jfrog/jfrog-client-go/actions/workflows/analysis.yml) passed.
- [ ] This pull request is on the dev branch.
- [ ] I used gofmt for formatting the code before submitting the pull request.
-----

### CODE REVIEW

**Comment by ns-kbhat** [comment]
@yahavi need your help in review on merge of this. Appreciate prompt response and help to unblock on this feature.

**Comment by ns-kbhat** [comment]
@RobiNino please check https://github.com/jfrog/jfrog-client-go/blob/1c352a93c23dcae0ef286e045854c68493af00c4/lifecycle/manager.go#L51

`lifecycle.RbPromotionParams` shows up as undefined.

**Comment by ns-kbhat** [comment]
@yahavi @RobiNino merge conflicts with `dev` branch are now resolved.

Appreciate prompt review and subsequent release, so we can update our tooling. Thanks in advance!

**Comment by ns-kbhat** [comment]
@yahavi @RobiNino when can I expect this PR to be reviewed?

I'm not clear what are the next steps for contributers - anything else required from my side?

**Comment by ns-kbhat** [comment]
> Thanks for your contribution @ns-kbhat ! I'll go over your PR tomorrow.
> 
> Nothing else is required, I'll leave any comments as part of the CR.

@RobiNino did you have a chance to review today?

**Review by RobiNino — CHANGES_REQUESTED** [review]
Hi @ns-kbhat ,
Left a few comments, let me know if I can assist in any way.
Please run `go fmt` when you're done.
Thanks again!

File: lifecycle/services/create.go [hunk 1]
```diff
@@ -69,10 +83,19 @@ type SourceBuildDetails struct {
...
+	Artifacts []ArtifactSource `json:"artifacts,omitempty"`
+}
+
 type CreateFromBuildsSource struct {
 	Builds []BuildSource `json:"builds,omitempty"`
 }
 
+type ArtifactSource struct {
+	Path   string `json:"path,omitempty"`
+	SHA256 string `json:"sha256,omitempty"`
```
[the remark this answers is not in the document]
Comment (ns-kbhat) [reply lifecycle/services/create.go]: somehow I missed using `Add suggestion to batch` button, maybe I was too focussed on making and validating with an actual distribution.

Will use batch accept in future PRs.

**Review by RobiNino — APPROVED** [review]

**Comment by RobiNino** [comment]
Much appreciated @ns-kbhat ! The changes will be included in our next release in the upcoming days

**Pull request merged**